From ed327e0da4c8671cabed8f4b2079efb90bb37cea Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:50 -0400
Subject: [ethernet] qlcnic: Use time_before()

Message-id: <1408626422-3941-127-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88672
O-Subject: [PATCH RHEL 7.1 126/138] qlogic: Use time_before()
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Manuel Schoelling <manuel.schoelling@gmx.de>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 2b3d9655119625cc102756395715292150a8980e

To be future-proof and for better readability the time comparisons are modified
to use time_before() instead of plain, error-prone math.

Signed-off-by: Manuel Schoelling <manuel.schoelling@gmx.de>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
index 7375817..5987f48 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
@@ -620,7 +620,7 @@ void qlcnic_prune_lb_filters(struct qlcnic_adapter *adapter)
  struct hlist_node *n;
  struct hlist_head *head;
  int i;
- unsigned long time;
+ unsigned long expires;
  u8 cmd;
 
  for (i = 0; i < adapter->fhash.fbucket_size; i++) {
@@ -628,8 +628,8 @@ void qlcnic_prune_lb_filters(struct qlcnic_adapter *adapter)
   hlist_for_each_entry_safe(tmp_fil, n, head, fnode) {
    cmd =  tmp_fil->vlan_id ? QLCNIC_MAC_VLAN_DEL :
         QLCNIC_MAC_DEL;
-   time = tmp_fil->ftime;
-   if (jiffies > (QLCNIC_FILTER_AGE * HZ + time)) {
+   expires = tmp_fil->ftime + QLCNIC_FILTER_AGE * HZ;
+   if (time_before(expires, jiffies)) {
     qlcnic_sre_macaddr_change(adapter,
          tmp_fil->faddr,
          tmp_fil->vlan_id,
@@ -647,8 +647,8 @@ void qlcnic_prune_lb_filters(struct qlcnic_adapter *adapter)
 
   hlist_for_each_entry_safe(tmp_fil, n, head, fnode)
   {
-   time = tmp_fil->ftime;
-   if (jiffies > (QLCNIC_FILTER_AGE * HZ + time)) {
+   expires = tmp_fil->ftime + QLCNIC_FILTER_AGE * HZ;
+   if (time_before(expires, jiffies)) {
     spin_lock_bh(&adapter->rx_mac_learn_lock);
     adapter->rx_fhash.fnum--;
     hlist_del(&tmp_fil->fnode);
-- 
1.7.1